engine/language-server/src/main/scala/org/enso/languageserver/libraries/LocalLibraryManager.scala

```diff
@@ -88,9 +85,10 @@ class LocalLibraryManager(
       )
     }
 
-    val libraryPath =
-      LocalLibraryProvider.resolveLibraryPath(librariesRoot, libraryName)
+    val libraryPath = librariesRoot.resolve(libraryName.name)
     if (Files.exists(libraryPath)) {
+      // TODO [RW] we could try finding alternative names (as directory name does not matter for local libraries), to find a free name
+      // This can be done as part of #1877
       throw new RuntimeException("Local library already exists")
     }
```

Comment:
The issue #1877 was closed some time ago as part of a cleanup, but it indeed remains unsolved - there are a few TODOs referencing it and indeed the error handling in that part is not fully developed but a shortcut.

It may be worth to re-open the issue. However, this part of the API is not yet being used by the IDE. I guess we may wait until it starts being used to re-visit these improvements. But let's keep in mind there is some missing functionality there that should be filled-in once this is actually used.